RFCs/RFC-001: Introspect streams.md

```diff
@@ -0,0 +1,62 @@
+# RFC-001: Introspect Streams
+
+**To be Reviewed By:** 11/19/2019
+
+**Authors:** Swapnil Bawaskar
+
+**Status:** **Active** | Accepted | Dropped | Superseded
+
+**Superseded by:** N/A
+
+**Related:** N/A
+
+
+## Problem
+When the streaming runtime is installed, riff users can create streams (backed by a stream provider e.g. kafka). Users can write functions that read from and write to these streams,
+but there is no easy mechanism to get events into the stream and look into the contents of the stream which are helpful while developing/demoing.
+
+### Anti-Goals
+We will only address this problem for development/demos, not production, so topics like auth/authz are out of scope for this document.
+
+## Solution
+### Ideal solution:
+The  is to provide users with only one cli (riff cli) that will enable users to send/receive messages to/from a stream (commands below). We will have to ask the users to port-forward the gateway service if they are running outside the cluster.
+
+#### User Impact
+We should introduce the following commands to the riff cli:
+- send a massage to the stream:
+
+    `riff streaming stream publish <stream-name> --payload <payload-as-string> --headers <headers-as-string>`  
+    content type has already been specified while creating the stream.
+
+- subscribe for messages from a stream:
+
+    `riff streaming stream subscribe <stream-name> --offset <long-offset>`  
+    this will display all the message from the given offset and then block to display further messages, essentially preventing user from entering subsequent commands.
+    The messages will be displayed as json with the following fields:
+    ``` 
+    {
+        "payload": "the users payload",
+        "content-type": "the content type of the message",
+        "headers": {"user provided header": "while publishing"}
+    }
+    ```
+
+### Interim solution:
```

Comment:
I'm really not at ease with the concept of interim solution vs ideal solution.
Either there is value to "solution B", which actually solves "problem B" and it should have its own RFC, including explanations about how problem B is different from problem A, or there is only one problem, and maybe several solutions, for which the RFC author can have (sometimes subjective) feelings about which is better than the other.

I think we fall in the former case, whereby @markfisher in particular sees a need for the low level binary, to be used in-cluster. Let's hear the usecases for that first, and then we can discuss the higher level solution (which I feel more familiar with). The "riff cli" solution may or may not build on this one, we'll see when we state the problem space then.

Reply:
fair point. I removed the "Interim Solution"